src/services/metadata-service.ts

```diff
@@ -67,12 +67,13 @@ export class MetadataService implements IMetadataService {
   /**
    * Retrieve genesis header fields from IPFS, store to the database.
    */
-  async fillFromIpfs(streamId: StreamID, options: AbortOptions = {}): Promise<void> {
-    const isPresent = await this.metadataRepository.isPresent(streamId)
-    if (isPresent) return // Do not perform same work of retrieving from IPFS twice
+  async fillFromIpfs(streamId: StreamID, options: AbortOptions = {}): Promise<GenesisFields> {
+    const storedFields = await this.metadataRepository.retrieve(streamId)
```

Comment:
@oed in order to get the count by did and model, i need to change this index hit to a database hit; it should only be for updates of existing streams that we already loaded previously.